### PR TITLE
Geocode with google maps geocode api

### PR DESCRIPTION
**Pull request by dride-for-sure**

### CODE REVIEW

**Review by Jothaka — COMMENTED** [review]
Lgtm,
as suggestion for tests I would probably try to create some helper methods to build the objects you keep using for every test as this really bloats up the testmethods.

**Comment by dride-for-sure** [comment]
> Lgtm,
> as suggestion for tests I would probably try to create some helper methods to build the objects you keep using for every test as this really bloats up the testmethods.

@Jothaka youre absolutely right!

File: backend/src/main/java/org/opensource/schulaltas/service/GeoService.java [hunk 1]
```diff
@@ -1,12 +1,60 @@
...
+                   .coordinates(
+                           Coordinates.builder()
+                                   .latitude( Double.parseDouble( geoLocationDto.getLatitude() ) )
+                                   .longitude( Double.parseDouble( geoLocationDto.getLongitude() ) )
+                                   .build() ).build() );
+  }
+  return Optional.empty();
+ }
+
+ private Optional<GeoResultsDto> geocode (Address address) {
```
[the remark this answers is not in the document]
Comment (dride-for-sure) [reply backend/src/main/java/org/opensource/schulaltas/service/GeoService.java:44]: RefactorFTW :)

File: backend/src/main/java/org/opensource/schulaltas/service/GeoService.java [hunk 2]
```diff
@@ -1,12 +1,60 @@
...
+  if ( geoResultsDto.isPresent() ) {
+   GeoLocationDto geoLocationDto =
+           geoResultsDto.get().getResults().get( 0 ).getGeometry().getGeocodeLocation();
+   return Optional.of(
+           GeoObject.builder()
+                   .coordinates(
+                           Coordinates.builder()
+                                   .latitude( Double.parseDouble( geoLocationDto.getLatitude() ) )
+                                   .longitude( Double.parseDouble( geoLocationDto.getLongitude() ) )
+                                   .build() ).build() );
```
[the remark this answers is not in the document]
Comment (dride-for-sure) [reply backend/src/main/java/org/opensource/schulaltas/service/GeoService.java:39]: Point! Looks like spaghetti code :(

**Comment by dride-for-sure** [comment]
Thanks for the Review @Jothaka ! 🚀

**Pull request merged**